libcove/lib/common.py

```diff
@@ -29,23 +29,29 @@
-                           'number': '\'{}\' is not a number. Check that the value {} doesn’t contain any characters other than 0-9 and dot (\'.\'). Number values should not be in quotes. ', # noqa
-                           'object': '\'{}\' is not a JSON object',
-                           'array': '\'{}\' is not a JSON array'}
+validation_error_template_lookup = {
+    'date': 'Date is not in the correct format. The correct format is YYYY-MM-DD.',
+    'date-time': 'Date is not in the correct format. The correct format is YYYY-MM-DDThh:mm:ssZ.',
+    'uri': 'Invalid uri found',
+    'string': '\'{}\' should be a string. Check that the value {} has quotes at the start and end. Escape any quotes in the value with \'\\\'', # noqa
+    'integer': '\'{}\' should be an integer. Check that the value {} doesn’t contain decimal points or any characters other than 0-9. Integer values should not be in quotes. ', # noqa
+    'number': '\'{}\' should be a number. Check that the value {} doesn’t contain any characters other than 0-9 and dot (\'.\'). Number values should not be in quotes. ', # noqa
+    'boolean': '\'{}\' should be a JSON boolean, \'true\' or \'false\'.', # noqa
+    'object': '\'{}\' should be a JSON object',
+    'array': '\'{}\' should be a JSON array. Check that value(s) appear within square brackets, [...]'}
 # These are "safe" html that we trust
 # Don't insert any values into these strings without ensuring escaping
 # e.g. using django's format_html function.
-validation_error_template_lookup_safe = {'date-time': 'Date is not in the correct format',
-                           'uri': 'Invalid \'uri\' found',
-                           'string': '<code>{}</code> is not a string. Check that the value {} has quotes at the start and end. Escape any quotes in the value with <code>\</code>', # noqa
-                           'integer': '<code>{}</code> is not a integer. Check that the value {} doesn’t contain decimal points or any characters other than 0-9. Integer values should not be in quotes. ', # noqa
-                           'number': '<code>{}</code> is not a number. Check that the value {} doesn’t contain any characters other than 0-9 and dot (<code>.</code>). Number values should not be in quotes. ', # noqa
-                           'object': '<code>{}</code> is not a JSON object',
-                           'array': '<code>{}</code> is not a JSON array'}
+validation_error_template_lookup_safe = {
+    'date': 'Date is not in the correct format. The correct format is YYYY-MM-DD.',
+    'date-time': 'Date is not in the correct format. The correct format is YYYY-MM-DDT00:00:00Z.',
+    'uri': 'Invalid uri found',
+    'string': '<code>{}</code> should be a string. Check that the value {} has quotes at the start and end. Escape any quotes in the value with <code>\</code>', # noqa
```

Comment:
The second instance of `{}` should also have `<code>` tags, to avoid ambiguity or confusion.

Reply:
The second `{}` is for a clause about nulls: https://github.com/OpenDataServices/lib-cove/blob/2c35429055091a0bf46a7a156c9d20cbf5d8c2fc/libcove/lib/common.py#L536-L549 but, this could be clearer.

Reply:
Thanks – I didn't read the formatting code, just the message templates. A short comment above the message templates about what values are substituted would improve clarity.